pkg/operator/ceph/object/user/controller.go

```diff
@@ -310,14 +310,20 @@ func (r *ReconcileObjectStoreUser) initializeObjectStoreContext(u *cephv1.CephOb
 	}
 
 	r.objContext = objContext
-	r.objContext.Endpoint = store.Status.Info["endpoint"]
-	if store.Status != nil && store.Status.Info["secureEndpoint"] != "" {
-		r.objContext.Endpoint = store.Status.Info["secureEndpoint"]
+	if store.Status == nil {
+		return errors.Errorf("unknown store status while waiting for store to initialize")
```

Comment:
Looking at [this original comment](https://github.com/rook/rook/issues/6916#issuecomment-810028168) again, I'm not sure that retrying will solve the issue. There must be scenarios where the status is not set on the store? At least if the object store fails to initialize (e.g. gets stuck on PGs not active+clean), this could fail on retry many times. At the same time, the object store issue has to resolve itself before the endpoint can be discovered so maybe there isn't a better answer.